src/app/pages/journal/__tests__/journal.page.spec.ts

```diff
@@ -0,0 +1,201 @@
+import { RouterTestingModule } from '@angular/router/testing';
+import { ComponentFactoryResolver } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { MockComponent } from 'ng-mocks';
+import { SearchResultTestSchema } from '@dvsa/mes-search-schema';
+import { JournalPage } from '../journal.page';
+import { AuthenticationProvider } from '../../../providers/authentication/authentication';
+import { AuthenticationProviderMock } from '../../../providers/authentication/__mocks__/authentication.mock';
+import { SlotSelectorProvider } from '../../../providers/slot-selector/slot-selector';
+import { DateTimeProvider } from '../../../providers/date-time/date-time';
+import { DateTimeProviderMock } from '../../../providers/date-time/__mocks__/date-time.mock';
+import { AppConfigProvider } from '../../../providers/app-config/app-config';
+import { AppConfigProviderMock } from '../../../providers/app-config/__mocks__/app-config.mock';
+import { AppComponent } from '../../../app.component';
+import { StoreModel } from '../../../shared/models/store.model';
+import { LoadingControllerMock } from '../../../../../mock/ionic-mocks/loading-controller.mock';
+import { MockAppComponent } from '../../../__mocks__/app.component.mock';
+import { JournalNavigationComponent } from '../components/journal-navigation/journal-navigation';
+import { selectVersionNumber } from '../../../../store/app-info/app-info.selectors';
+import { DateTime } from '../../../shared/helpers/date-time';
+import * as journalActions from '../../../../store/journal/journal.actions';
+import { MesError } from '../../../shared/models/mes-error.model';
+import { ErrorPage } from '../../error-page/error';
+import { ErrorTypes } from '../../../shared/models/error-message';
+import { MockBackButton } from './back-button.mock';
+import { MockPlatform } from './platform.mock';
+
+describe('JournalPage', () => {
+  let component: JournalPage;
+  let fixture: ComponentFixture<JournalPage>;
+  let mockBackButton; let
+    mockPlatform;
+  const platformReadySpy = jasmine.createSpy().and.returnValue(Promise.resolve());
+  let store$: MockStore;
+  const initialState = {
+    appInfo: {
+      versionNumber: '4.0',
+    },
+  } as StoreModel;
+  const mockNavParams = {
+    get: (param: string) => {
+      const data = {};
+      return data[param];
+    },
+  };
+
+  beforeEach(async(() => {
+    mockBackButton = new MockBackButton();
+    mockBackButton.subscribeWithPriority = jasmine.createSpy('subscribeWithPriority', () => {
+    });
+    mockPlatform = new MockPlatform();
+    mockPlatform.backButton = mockBackButton;
+    mockPlatform.ready = platformReadySpy;
+
+    const modalSpy = jasmine.createSpyObj('Modal', ['present']);
+    const modalControllerMock = jasmine.createSpyObj('ModalController', ['create']);
+    modalControllerMock.create.and.callFake(() => {
+      return modalSpy;
+    });
+
+    TestBed.configureTestingModule({
```

Comment:
This does not need to be inside a beforeEach

Reply:
following the guidance from ionic docs this is correct.

Reply:
![image](https://user-images.githubusercontent.com/48550267/108698032-9035f280-74fb-11eb-8061-5e2226385682.png)

Reply:
Pattern implemented initially then is wrong as currently none of the tests follow this format!

Reply:
if you use cli to create a component, the skeleton it creates is in this format. but we dont have time to refactor ALL unit testing if we can just drop working tests in.